src/shared/libs/iota/extendedApi.js

```diff
@@ -520,6 +513,37 @@ const checkAttachToTangleAsync = (node) => {
         });
 };
 
+/**
+ * Promisified version of iota.api.getTipInfo
+ *
+ * @method getTipInfoAsync
+ *
+ * @param {object} [settings]
+ *
+ * @returns {function(string): Promise<object>}
+ */
+const getTipInfoAsync = (settings) => (tailTransactionHash) => {
+    return fetch(settings.url, {
+        method: 'POST',
+        body: JSON.stringify({
+            command: 'getTipInfo',
+            tailTransaction: tailTransactionHash,
+        }),
+        headers: new Headers({
+            'Content-Type': 'application/json',
+            'X-IOTA-API-Version': IRI_API_VERSION,
+        }),
+    }).then((response) => {
+        return response.json().then((res) => {
```

Comment:
Should we check `response.ok` first before parsing JSON? We did that above for `checkAttachToTangleAsync`

Reply:
We are extracting error message from the parsed object `throw new Error(res.error);`, so I guess in this case it is fine to parse it before.

Reply:
Ok. The reason I ask is because if there's JSON parsing error, I don't think it'll be caught because the `throw` is inside `then`, but `response.json()` will run (and throw) before the `then` block